### PR TITLE
Improve test output

### PR DESCRIPTION
**Pull request by qoelet**
This sends additional output if a test fails (test name, contents of test input and golden). I would like to work on displaying just a diff. Would coloring the terminal output be overkill?

### CODE REVIEW

**Review by vrom911 — COMMENTED** [review]
These are very nice changes! 😍 

The coloured output will be very appreciated as this is a valuable investment that could help in testing processes, which is a very important part of the tools development. 
We are using the [`ansi-terminal`](http://hackage.haskell.org/package/ansi-terminal) library in our packages for such goals. Here you can see some useful pure functions that add colouring:
 https://github.com/kowainik/summoner/blob/master/src/Summoner/Ansi.hs#L88-L92

But probably you have any other ideas how to solve this, I'll be happy to discuss it with you!

**Comment by qoelet** [comment]
Thanks, I'll take a look at `ansi-terminal` and play with more specific output for test failures.

File: test/Test.hs [hunk 1]
```diff
@@ -23,11 +22,19 @@ main = do
...
+        else putStrLn "Some tests failed" >> exitFailure
   where
     compareWithGolden :: FilePath -> IO Bool
     compareWithGolden f = do
         let (name, _) = splitExtension f
         goldenContent <- readFile $ name <.> "golden"
         testContent   <- readFile f
-        pure $ goldenContent == testContent
+        if goldenContent == testContent
+          then return True
```
Comment (chshersh) [inline test/Test.hs]: Just a minor stylistic preference, but you can use `pure` instead of `return` (pure is shorter)

File: test/Test.hs [hunk 2]
```diff
@@ -23,11 +22,19 @@ main = do
...
+          else outputFailure name testContent goldenContent >> return False
```
Comment (chshersh) [inline test/Test.hs]: Alternatively, here you can use [`<$`](https://hackage.haskell.org/package/base-4.12.0.0/docs/Prelude.html#v:-60--36-) function from `Functor` and write this `if` branch in the following way:

```haskell
else False <$ outputFailure name testContent goldenContent
```

**Comment by chshersh** [comment]
@qoelet Could you paste couple screenshots of the test messages results?

**Comment by qoelet** [comment]
Successful
smuggler-success

Failure
smuggler-failure

**Review by vrom911 — APPROVED** [review]
Very cool! Thank you very much, @qoelet!

**Comment by qoelet** [comment]
:fireworks:

**Pull request merged**